🧹 Prepare erp order for correct er design

daniel-rose opened the pull request:
- add new table foo_erp_order_totals
- make specified fks temporally nullable
- fix deleting method
- increase versions in dandelion.json
- ...

sourcecube commented on bundles/one-time-password-rest-api/LICENSE:
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
2021 (;

sourcecube commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "dev-master as 1.0.0",
```
You can use 1.0.0, it was released 8 days ago (:

sourcecube commented on bundles/one-time-password-rest-api/src/FondOfOryx/Shared/OneTimePasswordRestApi/Transfer/one_time_password_rest_api.transfer.xml:
```diff
@@ -0,0 +1,11 @@
+        <property name="email" type="string"/>
+    </transfer>
+
+    <transfer name="OneTimePasswordResponse">
```
Should this be namespaced with `Rest` as well like the request? Or ist that how it's done in Spyker core?

sourcecube commented on bundles/one-time-password/LICENSE:
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
2021.. and for future record, these files should be in a separate PR

sourcecube commented on bundles/one-time-password/composer.json:
```diff
@@ -0,0 +1,36 @@
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "dev-master as 1.0.0",
```
`^1.0`

sourcecube commented on bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/OneTimePasswordRestApiConfig.php:
```diff
@@ -0,0 +1,14 @@
+    public const RESOURCE_ONE_TIME_PASSWORD = 'one-time-password';
+    public const CONTROLLER_ONE_TIME_PASSWORD = 'one-time-password-resource';
+
+    public const EMAIL_REQUIRED_ERROR_CODE = '1000';
+    public const EMAIL_REQUIRED_ERROR_DETAIL = 'The attribute Email is required.';
```
Nice! Maybe add a comment like .. "The code is not related to a system yet, might change later"

sourcecube commented on bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php:
```diff
@@ -0,0 +1,61 @@
+     *
+     * @return \Spryker\Glue\GlueApplication\Rest\JsonApi\RestResponseInterface
+     */
+    public function requestOneTimePasswordEmail(
+        RestRequestInterface $restRequest,
+        RestOneTimePasswordRequestAttributesTransfer $restOneTimePasswordRequestAttributesTransfer
+    ): RestResponseInterface {
+        if (!$restOneTimePasswordRequestAttributesTransfer->getEmail()) {
+            return $this->createEmailRequiredError();
+        }
+
+        return $this->restResourceBuilder
+            ->createRestResponse()
+            ->setStatus(204);
+    }
```
You could add a test for this to make sure that the error will be returned

sourcecube commented:
Good work 🎉

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+  ],
+  "require": {
+    "php": ">=7.2",
+    "spryker/glue-application": "^1.21.0"
```
Please use ^1.0.0

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
```
7.2 EOL

daniel-rose commented on composer.json:
```diff
@@ -19,6 +19,7 @@
     "spryker/customer": "^6.0.0 || ^7.0.0",
     "spryker/calculation": "^4.4.1",
     "spryker/country": "^3.0.0",
+    "spryker/glue-application": "^1.21.0",
```
Use ^1.0.0

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+    "spryker/glue-application": "^1.0.0"
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "^1.0.0",
```
use only `"fond-of-codeception/spryker": "^1.0",`

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "^1.0.0",
+    "mikey179/vfsstream": "^1.6",
+    "phpro/grumphp": "^0.14",
```
remove unused stuff

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+    "phpro/grumphp": "^0.14",
+    "sebastian/phpcpd": "^4.0",
+    "spryker/code-sniffer": "^0.11",
+    "spryker/development": "^3.6"
```
Why?

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+    "fond-of-codeception/spryker": "^1.0.0",
+    "mikey179/vfsstream": "^1.6",
+    "phpro/grumphp": "^0.14",
+    "sebastian/phpcpd": "^4.0",
```
Why?

daniel-rose commented on bundles/one-time-password-rest-api/composer.json:
```diff
@@ -0,0 +1,37 @@
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
`"Generated\\": "src/Generated/"`

julianzimmermann replied:
src/Generated/

daniel-rose commented on bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/OneTimePasswordRestApiDependencyProvider.php:
```diff
@@ -0,0 +1,42 @@
+    {
+        $container->set(
+            static::CLIENT_ZED_REQUEST,
+            function (Container $container) {
```
use static closure please.. (performance)

daniel-rose commented on bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/OneTimePasswordRestApiFactory.php:
```diff
@@ -0,0 +1,27 @@
+    /**
+     * @return \FondOfOryx\Client\OneTimePasswordRestApi\Dependency\Client\OneTimePasswordRestApiToZedRequestClientInterface
+     */
+    public function getZedRequestClient(): OneTimePasswordRestApiToZedRequestClientInterface
```
Can be protected...

daniel-rose commented on bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/Zed/OneTimePasswordRestApiStub.php:
```diff
@@ -0,0 +1,39 @@
+    public function requestOneTimePassword(
+        RestOneTimePasswordRequestAttributesTransfer $restOneTimePasswordRequestAttributesTransfer
+    ): RestOneTimePasswordResponseTransfer {
+        /**
+ * @var \Generated\Shared\Transfer\RestOneTimePasswordResponseTransfer $restOneTimePasswordResponseTransfer
+*/
```
CodeStyle...

daniel-rose commented on bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Controller/OneTimePasswordResourceController.php:
```diff
@@ -0,0 +1,29 @@
+class OneTimePasswordResourceController extends AbstractController
+{
+    /**
+     * @param \Spryker\Glue\GlueApplication\Rest\Request\Data\RestRequestInterface $restRequest
```
Add glue annotation please (for auto doc) --> https://documentation.spryker.com/docs/documenting-glue-api-resources#rest-methods

daniel-rose commented on bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php:
```diff
@@ -0,0 +1,75 @@
+
+        return $this->restResourceBuilder
+            ->createRestResponse()
+            ->setStatus(204);
```
Use const for this. (HttpFoundation from Symfony)

daniel-rose commented on bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php:
```diff
@@ -0,0 +1,75 @@
+    {
+        $restErrorMessageTransfer = new RestErrorMessageTransfer();
+
+        $restErrorMessageTransfer->setStatus(400)
```
Same

daniel-rose commented on bundles/one-time-password-rest-api/tests/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessorTest.php:
```diff
@@ -0,0 +1,149 @@
+
+        $this->restResponseMock->expects($this->atLeastOnce())
+            ->method('setStatus')
+            ->with(204)
```
Const from HttpFoundation

daniel-rose commented on bundles/one-time-password/composer.json:
```diff
@@ -0,0 +1,36 @@
+    }
+  ],
+  "require": {
+    "php": ">=7.2"
```
EOL for 7.2. Please use >= 7.3...

daniel-rose commented on bundles/one-time-password/composer.json:
```diff
@@ -0,0 +1,36 @@
+    "php": ">=7.2"
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "^1.0.0",
+    "mikey179/vfsstream": "^1.6",
+    "phpro/grumphp": "^0.14",
+    "sebastian/phpcpd": "^4.0",
+    "spryker/code-sniffer": "^0.11",
+    "spryker/development": "^3.6"
+  },
+  "minimum-stability": "dev",
+  "prefer-stable": true,
+  "autoload": {
+    "psr-4": {
+      "FondOfOryx\\": "src/FondOfOryx/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
Look at composer.json from one-time-password-rest-api...

daniel-rose reviewed (approved):
Keep it mind at dandelion.json

daniel-rose committed:
🧹 Prepare erp order for correct er design
- add new table foo_erp_order_totals
- make specified fks temporally nullable
- fix deleting method
- ...

daniel-rose committed:
Increase versions in dandelion.json

daniel-rose committed:
add phpcs fixes

daniel-rose committed:
WIP

daniel-rose committed:
WIP

daniel-rose committed:
add phpea fixes

daniel-rose committed:
wip

daniel-rose committed:
wip

daniel-rose committed:
updatde sub composer files

daniel-rose committed:
wip

daniel-rose committed:
Merge branch 'main' into feature/preparement-for-correct-er-design

# Conflicts:
#	dandelion.json
